corehq/apps/campaign/static/campaign/js/async_configurable_report.js

```diff
@@ -0,0 +1,428 @@
+) {
+    return function (o) {
+        var self = {};
+        self.widgetFilters = o.widgetFilters;
+        self.reportContent = o.reportContent;
+        self.filterForm = o.filterForm;
+        // self.loadingIssueModalElem = self.reportContent.find('#loadingReportIssueModal-' + o.reportId);
+        // self.loadingIssueModal = new bootstrap.Modal(self.loadingIssueModalElem.get(0));
+        self.issueAttempts = 0;
+        self.urlRoot = o.urlRoot;
+        
+        // Make sure urlRoot ends with a slash
+        if (self.urlRoot && !self.urlRoot.endsWith('/')) {
+            self.urlRoot += '/';
+        }
+        
+        self.hqLoading = null;
+        self.standardReport = o.standardReport;
+        self.filterRequest = null;
+        self.reportRequest = null;
+        self.queryIdRequest = null;
+        self.loaderClass = '.report-loading';
+        self.maxInputLimit = 4500;
+        self.reportId = o.reportId;
+        self.domain = o.domain || initialPageData.get('domain');
+
+        self.humanReadableErrors = {
+            400: gettext("Please check your Internet connection!"),
+            404: gettext("Report Not Found."),
+            408: gettext("Request timed out when rendering this report. This might be an issue with our servers" +
+                " or with your Internet connection. We encourage you to report an issue to CommCare HQ Support so we" +
+                " can look into any possible issues."),
+            500: gettext("Problem Rendering Report. Our error monitoring tools have noticed this and we are working quickly to" +
+                " resolve this issue as soon as possible. We encourage you to contact CommCare HQ Support" +
+                " if this issue persists for more than a few minutes. We appreciate any additional information" +
+                " you can give us about this problem so we can fix it immediately."),
+            502: gettext("Bad Gateway. Please contact CommCare HQ Support."),
+            503: gettext("CommCare HQ is experiencing server difficulties. We're working quickly to resolve it." +
+                " Thank you for your patience. We are extremely sorry."),
+            504: gettext("Gateway Timeout. Please contact CommCare HQ Support."),
+            'maxInputError': gettext("Your search term was too long. Please provide a shorter search filter"),
+        };
+
+        var loadFilters = function (data) {
+            self.filterRequest = null;
+            try {
+                // Hide loading indicator if it's still visible
+                self.widgetFilters.find('#hq-report-filters-' + self.reportId + ' .loading-filters').hide();
+                
+                // Check if filters data exists
+                if (data && data.filters) {
+                    self.widgetFilters.find('#hq-report-filters-' + self.reportId).html(data.filters);
+                    
+                    // Initialize any JavaScript components in the filters
```

Comment:
This is approximately what `hqwebapp/js/bootstrap5/widgets`.init does, what would you think of calling that instead?